File: Source/app/src/main/java/nl/endran/locust/maps/StringResourceMap.java
/*
 * Copyright (c) 2015 by David Hardy. Licensed under the Apache License, Version 2.0.
 */

package nl.endran.locust.maps;

import android.content.res.Resources;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;
import android.support.v4.app.Fragment;

import java.util.HashMap;
import java.util.Map;

import javax.inject.Inject;

import nl.endran.locust.R;
import nl.endran.locust.fragments.ResourcesFragment;
import nl.endran.locust.fragments.TerritoryFragment;
import nl.endran.locust.fragments.UnitsFragment;

public class StringResourceMap {

    @NonNull
    private final Resources resources;

    @NonNull
    private final Map<Class, Integer> classToStringResMap = new HashMap<>();

    @Inject
    public StringResourceMap(@NonNull final Resources resources) {
        this.resources = resources;

        classToStringResMap.put(UnitsFragment.class, R.string.units);
        classToStringResMap.put(ResourcesFragment.class, R.string.resources);
        classToStringResMap.put(TerritoryFragment.class, R.string.territory);
    }

    @NonNull
    public String getString(@StringRes final int stringRes) {
        return resources.getString(stringRes);
    }

    @StringRes
    public int getStringRes(Fragment fragment) {
        return classToStringResMap.get(fragment.getClass());
    }
}

File: Source/app/src/main/java/nl/endran/locust/maps/StringResourceMap.kt
/*
 * Copyright (c) 2015 by David Hardy. Licensed under the Apache License, Version 2.0.
 */

package nl.endran.locust.maps

import android.content.res.Resources
import android.support.annotation.StringRes
import android.support.v4.app.Fragment
import nl.endran.locust.R
import nl.endran.locust.fragments.ResourcesFragment
import nl.endran.locust.fragments.TerritoryFragment
import nl.endran.locust.fragments.UnitsFragment
import nl.endran.locust.game.units.Food
import nl.endran.locust.game.units.Nymph
import javax.inject.Inject

class StringResourceMap @Inject constructor(val resources: Resources) {
    val classToStringResMap = mapOf(
            UnitsFragment::class.java to R.string.units,
            ResourcesFragment::class.java to R.string.resources,
            TerritoryFragment::class.java to R.string.territory,

            Food::class.java to R.string.food,
            Nymph::class.java to R.string.nymph)

    fun getString(clazz: Class<*>): String {
        val stringId = classToStringResMap.getRaw(clazz)

        if (stringId != null) {
            return getString(stringId)
        } else {
            return clazz.simpleName
        }
    }

    fun getString(@StringRes stringRes: Int): String {
        return resources.getString(stringRes)
    }

    @StringRes
    fun getStringRes(fragment: Fragment): Int {
        return classToStringResMap[fragment.javaClass] ?: R.string.placeholder
    }
}
